docs/content/developer/tutorials/identity-workshop.mdx

```diff
@@ -0,0 +1,1417 @@
+5. ## Return Validation Result:
+    - Code:
+
+```rust
+output.push_str("\n🎉 All credentials in the VP are valid!");
+Ok(Json(ValidationResponse { success: true, output }))
+```
+- What it does: If all validations pass, logs a success message and returns a ValidationResponse with success: true and the detailed validation log (output) to the frontend.
+- Why it matters: The response informs the frontend whether the Verifiable Presentation and its Verifiable Credential are valid, updating the UI to show “SUCCESS” or “FAIL” and displaying the validation log.
+- Key concepts:
+    - ValidationResponse: A custom struct containing a boolean (success) and log (output).
+    - JSON Response: Serializes the result for HTTP communication with the React frontend.
+
+## Integration with the Workshop
+
+The verifier_validate function is triggered by the React frontend’s “Execute Step 4” button, which sends the VP JWT (from Step 3) to the /verifier/validate endpoint. It relies on:
+    - Input: The VP JWT (body.vp_jwt) from the frontend.
+    - Dependencies: get_read_only_client for network access and the Resolver for fetching DID documents.
+    - Prerequisites: The Student's DID (Step 1) and VC (Step 2) must exist, and the VP (Step 3) must be valid.
+
+The returned ValidationResponse updates the frontend UI, displaying the validation outcome and log.
+
+Output after a successful validation of the student’s Verifiable Presentation.
+![Create DID output](./images/VerifyVP2.png)
+    
+-==
+## Explanation:
+
+This completes the guided workshop, demonstrating the seamless and robust interaction between the React frontend and the Rust Axum backend across all four phases of the IOTA Identity flow.
+
+## Part 3: Running the Application
+
+1. Start the Backend:
+
+```Bash
+cd iota-identity-backend
+cargo run
+```
+
+This starts the Rust Axum server (e.g., at http://localhost:3000).
+
+2. Start the Frontend:
+
+```bash
+cd iota-identity-frontend
+npm run dev
+```
+
+This starts the React app (e.g., at http://localhost:5173).
+
+3. Interact with the UI:
+
+- Click  Create the Student's Decentralized Identifier.
+- Click  University the Verifiable Credential.
+- Click  Create the Verifiable Presentation.
+- Click  Verify the Verifiable Presentation.
+
+## Troubleshooting
+
+- Network Errors: Ensure the IOTA test network is running and `API_ENDPOINT` is set.
+- Dependency Issues: Run `cargo update` or `npm install` to resolve missing dependencies.
+- File Not Found: Verify that `holder_doc.json` exists after Step 1.
+- HTTP Errors: Check the backend logs for issues like invalid `packageId`.
+
+## What You’ve Accomplished
+
+You’ve built a full-stack IOTA Identity application:
+
+- Created a Decentralized Identifier for a student (Holder).
+- Issued a tamper-proof degree credential (VC).
+- Packaged the Verifiable Credential into a signed Verifiable Presentation.
+- Verified the Verifiable Presentation and Verifiable Credential on the IOTA network.
+
+## Workshop tasks
```

Comment:
Rename to 'Extension Tasks'
Ask users to build solutions to the alternative use cases or anything else we can think of
Ask them to present these solutions into the Discord channel for Identity to get feedback from the IOTA team

Reply:
Done